src/iccpd/src/mlacp_link_handler.c

```diff
@@ -0,0 +1,2319 @@
+    }
+
+    return client_fd;
+}
+
+int mclagd_ctl_sock_read(int fd, char *r_buf, int total_len)
+{
+    int read_len = 0;
+    int ret = 0;
+    struct timeval tv = { 0 };
+    fd_set read_fd;
+
+    while (read_len < total_len)
+    {
+        FD_ZERO(&read_fd);
+        FD_SET(fd, &read_fd);
+        tv.tv_sec = 5;
+        tv.tv_usec = 0;
+
+        switch ((ret = select(fd + 1, &read_fd, NULL, NULL, &tv)))
+        {
+            case -1:
+            /* error*/
+            case 0:
+                /* timeout*/
+                return -1;
+
+            default:
+                break;
+        }
+
+        if (FD_ISSET(fd, &read_fd))
+            ret = read(fd, r_buf + read_len, total_len - read_len);
+        if (ret <= 0)
+        {
+            return -1;
+        }
+        read_len += ret;
+    }
+
+    return read_len;
+}
+
+int mclagd_ctl_sock_write(int fd, char *w_buf, int total_len)
+{
+    int write_len = 0;
+    int ret = 0;
+
+    while (write_len < total_len)
+    {
+        ret = write(fd, w_buf + write_len, total_len - write_len);
+        if (ret <= 0)
+        {
+            return 0;
+        }
+        write_len += ret;
+    }
+
+    return write_len;
+}
+
+void mclagd_ctl_handle_dump_state(int client_fd, int mclag_id)
+{
+    char * Pbuf = NULL;
+    char buf[512] = { 0 };
+    int state_num = 0;
+    int ret = 0;
+    struct mclagd_reply_hdr *hd = NULL;
+    int len_tmp = 0;
+
+    ret = iccp_mclag_config_dump(&Pbuf, &state_num, mclag_id);
+    ICCPD_LOG_WARN(__FUNCTION__, "state_num = %d", state_num);
+    if (ret != EXEC_TYPE_SUCCESS)
+    {
+        len_tmp = sizeof(struct mclagd_reply_hdr);
+        memcpy(buf, &len_tmp, sizeof(int));
+        hd = (struct mclagd_reply_hdr *)(buf + sizeof(int));
+        hd->exec_result = ret;
+        hd->info_type = INFO_TYPE_DUMP_STATE;
+        hd->data_len = 0;
+        mclagd_ctl_sock_write(client_fd, buf, MCLAGD_REPLY_INFO_HDR);
+
+        if (Pbuf)
+            free(Pbuf);
+
+        return;
+    }
+    hd = (struct mclagd_reply_hdr *)(Pbuf + sizeof(int));
+    hd->exec_result = EXEC_TYPE_SUCCESS;
+    hd->info_type = INFO_TYPE_DUMP_STATE;
+    hd->data_len = state_num * sizeof(struct mclagd_state);
+
+    len_tmp = (hd->data_len + sizeof(struct mclagd_reply_hdr));
+    memcpy(Pbuf, &len_tmp, sizeof(int));
+    mclagd_ctl_sock_write(client_fd, Pbuf, MCLAGD_REPLY_INFO_HDR + hd->data_len);
+
+    if (Pbuf)
+        free(Pbuf);
+
+    return;
+}
+
+void mclagd_ctl_handle_dump_arp(int client_fd, int mclag_id)
+{
+    char * Pbuf = NULL;
+    char buf[512] = { 0 };
+    int arp_num = 0;
+    int ret = 0;
+    struct mclagd_reply_hdr *hd = NULL;
+    int len_tmp = 0;
+
+    ret = iccp_arp_dump(&Pbuf, &arp_num, mclag_id);
+    if (ret != EXEC_TYPE_SUCCESS)
+    {
+        len_tmp = sizeof(struct mclagd_reply_hdr);
+        memcpy(buf, &len_tmp, sizeof(int));
+        hd = (struct mclagd_reply_hdr *)(buf + sizeof(int));
+        hd->exec_result = ret;
+        hd->info
```

Comment:
The size should be set to MCLAGD_REPLY_INFO_HDR to account for 4-byte overall length

Reply:
> The size should be set to MCLAGD_REPLY_INFO_HDR to account for 4-byte overall length

@lisahnguyen, Thanks, we will fix it ASAP.